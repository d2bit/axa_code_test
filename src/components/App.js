import React from 'react'
import { injectGlobal } from 'styled-components'
import GnomesContainer from '../containers/GnomesContainer'

const App = () => {
  injectGlobal`
    body {
      margin: 0;
      font-family: sans-serif;
    }
  `
  return <GnomesContainer />
}

export default App
