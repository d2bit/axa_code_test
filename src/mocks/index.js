import data from './data.json'

const population = Object.values(data)[0]
const populationCount = population.length

export function getPopulation(size=5) {
  if (size >= populationCount) {
    return population
  }
  return population.slice(0, size)
}

export function getGnome(index) {
  if (index && index >= 0 && index < populationCount) {
    return population[index]
  }
  const randomIndex = Math.floor(Math.random() * populationCount)
  return population[randomIndex]
}

export function getResponseData() {
  return data
}

const Helper = {
  getPopulation,
  getGnome,
  getResponseData,
}

export default Helper
