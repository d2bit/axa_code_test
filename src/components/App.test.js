import React from 'react'
import { render } from 'react-testing-library'
import App from './App'
import DataDownloader from '../libs/dataDownloader'
import { getResponseData } from '../mocks'
jest.mock('../libs/dataDownloader')

describe('App', () => {
  it('renders without crashing', () => {
    DataDownloader.get.mockResolvedValue(getResponseData())

    const { component } = render(<App />)

    expect(component).toMatchSnapshot()
  })
})
