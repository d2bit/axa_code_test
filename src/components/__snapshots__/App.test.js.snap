// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div>
  <div
    class="sc-bZQynM koJOpM"
  >
    <label
      class="sc-gzVnrw ijciRG"
      for="GnomeFilter"
    >
      Filter
    </label>
    <input
      class="sc-htoDjs icjMxa"
      data-testid="GnomeFilter"
      name="GnomeFilter"
      value=""
    />
  </div>
  <div
    class="sc-EHOje cFMLaD"
    data-testid="GnomeList"
  >
    <article
      class="sc-bdVaJa ibopQg"
      data-testid="Gnome-0"
    >
      <h1
        class="sc-bwzfXH ktvMFN"
        data-testid="name"
      >
        Tobus Quickwhistle
      </h1>
      <img
        alt="Tobus Quickwhistle"
        class="sc-htpNat bxYIbZ"
        data-testid="thumbnail"
        src="http://www.publicdomainpictures.net/pictures/10000/nahled/thinking-monkey-11282237747K8xB.jpg"
        width="200px"
      />
      <div
        class="sc-bxivhb AhLTN"
      >
        <dl
          class="sc-VigVT edbqte"
        >
          <dt
            class="sc-jTzLTM bsobgB"
          >
            Friends #
          </dt>
          <dd
            class="sc-fjdhpX gLVTub"
          >
            2
          </dd>
        </dl>
        <dl
          class="sc-jzJRlG EHqGR"
        >
          <dt
            class="sc-cSHVUG lhbslC"
          >
            Professions #
          </dt>
          <dd
            class="sc-kAzzGY fqaLFK"
          >
            6
          </dd>
        </dl>
        <button
          class="sc-ifAKCX dFiLvw"
        >
          OPEN
        </button>
      </div>
    </article>
    <article
      class="sc-bdVaJa ibopQg"
      data-testid="Gnome-1"
    >
      <h1
        class="sc-bwzfXH ktvMFN"
        data-testid="name"
      >
        Fizkin Voidbuster
      </h1>
      <img
        alt="Fizkin Voidbuster"
        class="sc-htpNat bxYIbZ"
        data-testid="thumbnail"
        src="http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg"
        width="200px"
      />
      <div
        class="sc-bxivhb AhLTN"
      >
        <dl
          class="sc-chPdSV bFTptm"
        >
          <dt
            class="sc-kgoBCf jkhvua"
          >
            Friends #
          </dt>
          <dd
            class="sc-kGXeez cjREQU"
          >
            0
          </dd>
        </dl>
        <dl
          class="sc-kpOJdX iWMmLY"
        >
          <dt
            class="sc-dxgOiQ iWMJYb"
          >
            Professions #
          </dt>
          <dd
            class="sc-ckVGcZ jSkjkd"
          >
            6
          </dd>
        </dl>
        <button
          class="sc-ifAKCX dFiLvw"
        >
          OPEN
        </button>
      </div>
    </article>
    <article
      class="sc-bdVaJa ibopQg"
      data-testid="Gnome-2"
    >
      <h1
        class="sc-bwzfXH ktvMFN"
        data-testid="name"
      >
        Malbin Chromerocket
      </h1>
      <img
        alt="Malbin Chromerocket"
        class="sc-htpNat bxYIbZ"
        data-testid="thumbnail"
        src="http://www.publicdomainpictures.net/pictures/30000/nahled/maple-leaves-background.jpg"
        width="200px"
      />
      <div
        class="sc-bxivhb AhLTN"
      >
        <dl
          class="sc-jKJlTe hPVijE"
        >
          <dt
            class="sc-eNQAEJ cbkkjm"
          >
            Friends #
          </dt>
          <dd
            class="sc-hMqMXs ceJWUR"
          >
            1
          </dd>
        </dl>
        <dl
          class="sc-kEYyzF blYzxA"
        >
          <dt
            class="sc-kkGfuU cHHKFH"
          >
            Professions #
          </dt>
          <dd
            class="sc-iAyFgw hCWBoI"
          >
            3
          </dd>
        </dl>
        <button
          class="sc-ifAKCX dFiLvw"
        >
          OPEN
        </button>
      </div>
    </article>
  </div>
</div>
`;
